src/repositories/organisations/port.js

```diff
@@ -1,22 +1,48 @@
 /**
- * @typedef {Object} Accreditation
+ * @typedef {Object} AccreditationBase
  * @property {string} id
- * @property {number} accreditationNumber
  * @property {string} material
  * @property {string} wasteProcessingType
+ * @property {Date} formSubmissionTime
+ * @property {string} submittedToRegulator
  */
 
 /**
- * @typedef {Object} Registration
+ * @typedef {AccreditationBase & {status: 'approved'|'suspended', accreditationNumber: string, validFrom: Date, validTo: Date}} AccreditationApproved
+ */
+
+/**
+ * @typedef {AccreditationBase & {status: 'created'|'rejected'|'archived', accreditationNumber?: string, validFrom?: Date, validTo?: Date}} AccreditationOther
```

Comment:
I'm wondering if `validFrom` and `validTo` are required for accreditations in these statuses...

When are those data points created (and who creates them)?

If I had an accreditation that went from `created > approved > archived` the from/to dates would (presumably) be set. Is there any way to get to `archived` without those data points?

(Perhaps a better question is "what would break if we had an accreditation in the `created` / `rejected` / `archived` status that did not have `validFrom / validTo` dates?")